test/e2e/openshift/openshift_test.go

```diff
@@ -171,4 +175,36 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		Expect(err).NotTo(HaveOccurred())
 		Expect(running).To(Equal(true))
 	})
+
+	It("should not deploy infrastructure components on compute nodes", func() {
+		namespaces := []string{
+			"openshift-",
+			"kube-",
+			"default",
+		}
+
+		list, err := pod.GetAll("")
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, pod := range list.Pods {
```

Comment:
```
nextPod:
		for _, pod := range list.Pods {
			// skip if we find a pod that is a daemonset
			for _, ref := range pod.Metadata.OwnerReferences {
				if ref.Kind == "DaemonSet" {
					continue nextPod
				}
			}
```

^ is better because (a) it would handle multiple ownerreferences if they ever happened for some reason, and (b) testing for a non-nil slice does not imply that the slice has an element (nb you can use len > 0 for that since len(nil slice) is defined to be 0)